src/NzbDrone.Core/Download/Clients/DownloadStation/Proxies/DiskStationProxyBase.cs

```diff
@@ -1,58 +1,86 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Linq;
 using System.Net;
 using NLog;
+using NzbDrone.Common.Cache;
 using NzbDrone.Common.Http;
 using NzbDrone.Common.Serializer;
 using NzbDrone.Core.Download.Clients.DownloadStation.Responses;
 
 namespace NzbDrone.Core.Download.Clients.DownloadStation.Proxies
 {
-    public abstract class DiskStationProxyBase
+    public interface IDiskStationProxy
     {
-        private static readonly Dictionary<DiskStationApi, string> Resources;
+        IEnumerable<int> GetApiVersion(DownloadStationSettings settings);
+    }
+
+    public abstract class DiskStationProxyBase : IDiskStationProxy
+    {
+        private readonly Dictionary<DiskStationApi, DiskStationApiInfo> _resources;
 
         private readonly IHttpClient _httpClient;
         protected readonly Logger _logger;
+        private ICached<int> _cache;
         private bool _authenticated;
-
-        static DiskStationProxyBase()
+        
+        public DiskStationProxyBase(IHttpClient httpClient,
+                                    Logger logger,
+                                    ICacheManager cacheManager)
```

Comment:
We generally do Logger as last parameter and new stuff from derived classes get prepended.